ArjunaJTA/tomcat/src/main/java/org/jboss/narayana/jta/tomcat/NarayanaJtaLifecycleListener.java

```diff
@@ -0,0 +1,135 @@
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.narayana.jta.tomcat;
+
+import com.arjuna.ats.arjuna.common.CoreEnvironmentBeanException;
+import com.arjuna.ats.arjuna.common.arjPropertyManager;
+import com.arjuna.ats.arjuna.common.recoveryPropertyManager;
+import com.arjuna.ats.arjuna.coordinator.TransactionReaper;
+import com.arjuna.ats.arjuna.coordinator.TxControl;
+import com.arjuna.ats.arjuna.recovery.RecoveryManager;
+import com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule;
+import com.arjuna.ats.internal.arjuna.recovery.ExpiredTransactionStatusManagerScanner;
+import com.arjuna.ats.internal.jta.recovery.arjunacore.JTANodeNameXAResourceOrphanFilter;
+import com.arjuna.ats.internal.jta.recovery.arjunacore.JTATransactionLogXAResourceOrphanFilter;
+import com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule;
+import com.arjuna.ats.jdbc.TransactionalDriver;
+import com.arjuna.ats.jta.common.jtaPropertyManager;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
+ */
+public class NarayanaJtaLifecycleListener implements ServletContextListener {
+
+    private static final Logger LOGGER = Logger.getLogger(NarayanaJtaLifecycleListener.class.getSimpleName());
+
+    private static final String DEFAULT_NODE_IDENTIFIER = "1";
+
+    private static final List<String> DEFAULT_RECOVERY_MODULES = Arrays.asList(AtomicActionRecoveryModule.class.getName(),
+            XARecoveryModule.class.getName());
+
+    private static final List<String> DEFAULT_ORPHAN_FILTERS = Arrays
+            .asList(JTATransactionLogXAResourceOrphanFilter.class.getName(), JTANodeNameXAResourceOrphanFilter.class.getName());
+
+    private static final List<String> DEFAULT_EXPIRY_SCANNERS = Arrays
+            .asList(ExpiredTransactionStatusManagerScanner.class.getName());
+
+    @Override
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+        LOGGER.info("Initializing Narayana");
```

Comment:
Is there a precedent? We don't want to be the only subsystem that does this. If there is a precedent (either to info or trace) lets go with that. Otherwise we can use the precedent in WFLY.